app/assets/javascripts/components/settings/views/add_admin_button.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import Popover from '../../common/popover.jsx';
+import PopoverExpandable from '../../high_order/popover_expandable.jsx';
+import AddAdminForm from '../containers/add_admin_form_container';
+
+class AddAdminButton extends React.Component {
+  constructor() {
+    super();
+    this.getKey = this.getKey.bind(this);
+    this.render = this.render.bind(this);
+  }
+
+  getKey() {
+    return `add_${this.props.source}_button`;
+  }
+
+  closePopover() {
+    // what goes here???
+  }
+
+  render() {
+    const form = <AddAdminForm handlePopoverClose={this.closePopover} />;
```

Comment:
For components wrapped in `PopoverExpandable`, `this.props.open` is responsible for both opening and closing. Just pass this.props.open directly, and make sure you pass it an event to when you call it from `AddAdminForm`.